vmm/src/vm.rs

```diff
@@ -427,17 +412,6 @@ impl VmOps for VmOpsHandler {
 
     #[cfg(target_arch = "x86_64")]
     fn pio_write(&self, port: u64, data: &[u8]) -> result::Result<(), HypervisorVmError> {
-        use pci::{PCI_CONFIG_IO_PORT, PCI_CONFIG_IO_PORT_SIZE};
-
-        if (PCI_CONFIG_IO_PORT..(PCI_CONFIG_IO_PORT + PCI_CONFIG_IO_PORT_SIZE)).contains(&port) {
-            self.pci_config_io.lock().unwrap().write(
-                PCI_CONFIG_IO_PORT,
-                port - PCI_CONFIG_IO_PORT,
-                data,
-            );
-            return Ok(());
-        }
-
```

Comment:
@michael2012z Here is the optimisation that is removed. I suggested that @sboeuf not use `VmmOps` on KVM and instead do this optimisation in the exit handler inside the vCPU code instead.

Reply:
Thanks for pointing that.

I think it's better to close this PR now. When @sboeuf is back, he may need to add more commits and it's better to make a new PR from his branch then. 

I will keep the commits of this PR in my [branch](https://github.com/michael2012z/cloud-hypervisor/tree/switch_cpu_device_modified) unchanged. Then please cherry-pick the work there.

Reply:
Thank you for taking care of the GIC and the deadlock @michael2012z.
I'm gonna work on re-enabling the optimization and I'll send a PR when that's done.